app/models/mixins/relationship_mixin.rb

```diff
@@ -664,10 +733,14 @@ def remove_all_relationships(*rels)
   end
 
   def is_descendant_of?(obj)
+    return call_ancestry_method(:descendant_of?, obj) if use_ancestry?
+
     ancestor_ids.include?([obj.class.base_class.name, obj.id])
   end
 
   def is_ancestor_of?(obj)
+    return obj.with_relationship_type(relationship_type) { obj.descendant_of?(self) } if use_ancestry?
```

Comment:
looking up descendant ids is a database call but looking up ancestors is not.

so calling is_descendant_of requires fewer queries.